pkg/handler/crud_test.go

```diff
@@ -42,6 +42,36 @@ func TestCrudFlags(t *testing.T) {
 		assert.Equal(t, "some_random_flag_key", res.(*flag.CreateFlagOK).Payload.Key)
 	})
 
+	t.Run("it should be able to create one simple flag", func(t *testing.T) {
+		res = c.CreateFlag(flag.CreateFlagParams{
+			Body: &models.CreateFlagRequest{
+				Description: util.StringPtr("simple flag"),
+				Key:         "simple_flag_key",
+				Template:    "simple",
+			},
+		})
+		assert.NotZero(t, res.(*flag.CreateFlagOK).Payload.ID)
+		assert.Equal(t, "simple_flag_key", res.(*flag.CreateFlagOK).Payload.Key)
```

Comment:
I could assert more on the return payload here, as I realized it should return variants/segments as part of it meaning we can do full validation

Reply:
actually looks like payload isnt correct - it does save segment/variant/etc but doesnt return in payload.. will explore but if you have guidance lmk

Reply:
fixed this but would appreciate feedback if the way im managing the arrays is correct